### PR TITLE
Added the ability to import and export the ciphered private key

### LINKED ISSUE #24: [Request] Add Encrypted exports + imports
To enhance the security of MPW, we should start *discouraging the use of naked WIF private keys* in day-to-day use, by allowing full use of our existing AES-GCM encrypted MPW keys.

In theory this means an update to the password manager which either:
- **Easier:** A pattern to recognise AES-GCM keys and switch to the 'Enter Password' mode.
- **Harder:** Improve our AES keys by prefixing them with a very specific byte serial, so we can guarantee an input is 'WIF' or 'AES-WIF', this also requires **carefully** updating existing keys, with full backwards-compat, and then again switching to 'Enter Password' mode upon key import.

Using a unique byte serial prefix gives MPW keys a distinct verifiability (a deterministic key type, so we can verify: prefix, salt, IV, and encrypted bytes), and thus is a much preferred method to 'guessing' if a key is encrypted.

After this, we should add a dedicated / refined UI to 'Export' encrypted keys either to the Clipboard or directly to a custom MPW file, which could be used as an import file in addition to the copy/paste import method.

## TL;DR recap:
A developer here, should:
- Define and write a method of detecting if a key if plain WIF or AES-WIF.
- Import said key to storage, switch to 'Unlock Wallet' mode.
- Create an Export UI which supports two options: Clipboard or File.

If anyone picks this task up, please ping me here, I'll gladly assist, and a Labs bounty is attached as per the GitHub labels.

### PR DESCRIPTION
**Pull request by Shangaman**
Abstract:
I added the possibility to import/export the wallet with the internal ciphered private key.

The input box recognize from the first character (which is the non-base58 character +) if the private key is ciphered or not and in case it allows to insert the password.

Closes #24

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -192,7 +192,7 @@ <h5>
                   <div class="col-md-11 pivx-feature-interior">
 
                     <div class="col-md-12 feature-icon">
-                      <img src="https://pivx.org/build/images/content/img_privacy.png" alt="PIVX Privacy" class="smaller-feature">
+                      <img src="https://pivx.org/build/images/content/img_privacy.png" alt="P1IVX Privacy" class="smaller-feature">
```
Comment (JSKitty) [inline index.html:195]: nit: this was possibly an accident?

File: index.html [hunk 2]
```diff
@@ -769,14 +775,42 @@ <h4 id="signed-raw">Signed Raw Transaction</h4>
...
+        domPrivateCipheredTxt.style.display='none';
+        domPublicQr.style.display ='none'
+        domPrivateQr.style.display='block'
+        domPrivateCipheredQr.style.display='none'
+        break;
+        
+      }
     }
-
+
```
Comment (JSKitty) [inline index.html:813]: nit: unnecessary whitespace

File: scripts/misc.js [hunk 3]
```diff
@@ -36,6 +36,7 @@ const to_b58 = function (B) {
     var d = [],    //the array for storing the stream of base58 digits
         s = "",    //the result string variable that will be returned
         i,         //the iterator variable for the byte input
+
```
Comment (JSKitty) [inline scripts/misc.js:39]: nit: unnecessary whitespace

**Review by JSKitty — CHANGES_REQUESTED** [review]
The majority of the code is great! Some minor nits left, and I will do any style cleanup post-merge.

There is one **major** thing I've realised during review + after running some tests: MPW cipherkeys can be recognised with the below practice:
- We know WIF keys are fixed-length.
- We know the PBKDF2 password key is fixed-length.
- We know AES output natively contain non-Base58 chars (i.e: `/, 0, O, I, l`).

I conclude that our cipherkeys are **always** a fixed 1024 bits, or 128 bytes (`.length`), and will always include non-Base58 without the manual addition of a `+` symbol.

If you change the implementation to check for an input length of `128`, and check for non-Base58 using [my existing code](https://github.com/PIVX-Labs/MyPIVXWallet/blob/1d37032a91fda2f4f8050ea0158493592af01cb7/index.html#L875), we can have a very clean and safe implementation, and then we'll merge! :rocket:

File: index.html (hunk 1 above)
Comment (Shangaman) [reply index.html:195]: > The majority of the code is great! Some minor nits left, and I will do any style cleanup post-merge.
> 
> There is one **major** thing I've realised during review + after running some tests: MPW cipherkeys can be recognised with the below practice:
> 
> * We know WIF keys are fixed-length.
> * We know the PBKDF2 password key is fixed-length.
> * We know AES output natively contain non-Base58 chars (i.e: `/, 0, O, I, l`).
> 
> I conclude that our cipherkeys are **always** a fixed 1024 bits, or 128 bytes (`.length`), and will always include non-Base58 without the manual addition of a `+` symbol.
> 
> If you change the implementation to check for an input length of `128`, and check for non-Base58 using [my existing code](https://github.com/PIVX-Labs/MyPIVXWallet/blob/1d37032a91fda2f4f8050ea0158493592af01cb7/index.html#L875), we can have a very clean and safe implementation, and then we'll merge! rocket

Ok I agree that the + can be removed and looking at the length is better (which will fixed and different for plain and ciphered). However I think that it is possible (even if very unlikely) that the ciphered message in base64 doesn't contain any non-base58 characters.
So I will just check for length, what you think?

Comment (JSKitty) [reply index.html:195]: That should also work fine! Since it should be impossible for a PIVX WIF and an MPW cipherWIF to be the same length, actually. :+1:

**Review by JSKitty — APPROVED** [review]
tACK! Smooth cipherWIF export/import, really nice to have this finally, I'll continue with any cleanup, great job! :rocket:
